cmd/launcher/svc_windows.go

```diff
@@ -231,6 +232,15 @@ func (w *winSvc) Execute(args []string, r <-chan svc.ChangeRequest, changes chan
 					"change_request", fmt.Sprintf("%+v", c),
 				)
 			}
+		case <-runLauncherResults:
```

Comment:
_This_ is interesting. I wonder if it's a fix for the occasional timeout issues we see from service manager.

Reply:
Unfortunately this is the part that is the trickiest and that I'm the least certain about 😞 

Basically, no matter which option I try, when we exit the service like this, the service manager won't restart it. I'm not sure why.

Reply:
Okay, I think I figured it out. Basically when launcher shuts down and we want service manager to restart it, we should NOT send `service stopped` to the status changes channel. Prior to this PR, we were sending that status update -- but now my best guess is that because we sent `service stopped` to the status changes channel and _then_ immediately called `os.Exit`, that change didn't actually have time to propagate through.

Reply:
I think this change makes sense but per the [docs](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_failure_actions_flag) we should be able to do one or the other now and still get the service control manager restart (e.g. either set non-zero exit code _or_ don't set service stopped state). So just in the interest of making sure I understand here - if you set the service state to stopped, and then return `windows.ERROR_EXCEPTION_IN_SERVICE`, we don't see a restart?

Reply:
I'd expect based on the behavior you're describing that the exit code is ending up as `ERROR_SUCCESS (0)`, _and_ the service is stopped from service manager perspective

Reply:
> So just in the interest of making sure I understand here - if you set the service state to stopped, and then return windows.ERROR_EXCEPTION_IN_SERVICE, we don't see a restart?

Correct!

> I'd expect based on the behavior you're describing that the exit code is ending up as ERROR_SUCCESS (0), and the service is stopped from service manager perspective

That's largely what I was seeing before making the change in https://github.com/kolide/launcher/pull/1693/commits/63c2f1a02e9fce5dc157c6817424b903c97ebf9c -- no matter what exit code I returned, if I set the service stopped state, the service would be stopped and remain stopped (never restarted), and querying via `Get-CimInstance` would show `ExitCode` 0 as well.

Reply:
Gotcha, well regardless of what is masking that exit code to service manager - not setting the stopped state seems like a safer route towards ensuring restart anyway